rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -716,12 +746,26 @@ rmw_ret_t rmw_destroy_node(rmw_node_t * node)
 
   Participant * participant = impl->participant;
 
+  // Begin deleting things in the same order they were created in rmw_create_node().
   std::pair<StatefulReader *, StatefulReader *> EDPReaders = participant->getEDPReaders();
-  EDPReaders.first->setListener(nullptr);
-  delete (impl->secondarySubListener);
-  EDPReaders.second->setListener(nullptr);
-  delete (impl->secondaryPubListener);
+  if (!EDPReaders.first->setListener(nullptr)) {
+    RMW_SET_ERROR_MSG("failed to unset EDPReader listener");
+    return RMW_RET_ERROR;
+  }
+  delete impl->secondarySubListener;
+  if (!EDPReaders.second->setListener(nullptr)) {
+    RMW_SET_ERROR_MSG("failed to unset EDPReader listener");
+    return RMW_RET_ERROR;
+  }
+  delete impl->secondaryPubListener;
 
+  if (node) {
+    free(const_cast<char *>(node->name));
+    free(const_cast<char *>(node->namespace_));
```

Comment:
But `node` is deallocated on the next line, so I didn't. I don't know what debugging will happen between here and there.

Either way: But `node` is deallocated on the next line, so I didn't. I don't know what debugging will happen between here and there.

Either way: 79c4860